src/interfaces/parser.ts

```diff
@@ -56,7 +56,7 @@ export type DefaultContext<T> = {
 }
 
 export type FlagDefault<T, P = CustomOptions> = T | ((context: DefaultContext<P & OptionFlag<T, P>>) => Promise<T>)
-export type FlagDefaultHelp<T, P = CustomOptions> = T | ((context: DefaultContext<P & OptionFlag<T, P>>) => Promise<string | undefined>)
+export type FlagDefaultHelp<T, P = CustomOptions> = T | ((context: Omit<DefaultContext<P & OptionFlag<T, P>>, 'flags'>) => Promise<string | undefined>)
```

Comment:
Omitting flags because that property is meant to hold the flags that the user entered. Does not make sense in help to build cache context.